SPEC.md

```diff
@@ -0,0 +1,195 @@
+        }
+    ],
+
+    // This field should be applied to the Container's OCI spec if any of the
+    // devices defined above are requested on the CLI
+    "containerSpec": [
+        {
+            "devices": [ (optional)
+                {
+                    "hostPath": "<path>",
+                    "containerPath": "<path>",
+
+                    // Cgroups permissions of the device, candidates are one or more of
+                    // * r - allows container to read from the specified device.
+                    // * w - allows container to write to the specified device.
+                    // * m - allows container to create device files that do not yet exist.
+                    "permissions": "<permissions>" (optional)
+                }
+            ]
+            "mounts": [ (optional)
+                {
+                    "hostPath": "<source>",
+                    "containerPath": "<destination>",
+                    "options": "<OCI Mount Options>", (optional)
+                }
+            ],
+            "hooks": [ (optional)
+                {
+                    "hookName": "<hookName>",
+                    "path": "<path>",
+                    "args": ["<arg>", "<arg>"], (optional)
+                    "env":  [ "<envName>=<envValue>"], (optional)
+                    "timeout": <int> (optional)
+                }
+            ]
+        }
+    ]
+}
+```
+
+#### Specification version
+
+* `cdiVersion` (string, REQUIRED) MUST be in [Semantic Version 2.0](https://semver.org) and specifies the version of the CDI specification used by the vendor.
+
+#### Kind
+
+* `kind` (string, REQUIRED) field specifies a label which uniquely identifies the device vendor.
+  It can be used to disambiguate the vendor that matches a device, e.g: `docker/podman run --device vendor.com/device=foo ...`.
+    * The `kind` and `kindShort` labels have two segments: a prefix and a name, separated by a slash (/).
+    * The name segment is required and must be 63 characters or less, beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (\_), dots (.), and alphanumerics between.
+    * The prefix must be a DNS subdomain: a series of DNS labels separated by dots (.), not longer than 253 characters in total, followed by a slash (/).
+    * Examples (not an exhaustive list):
+      * Valid: `vendor.com/foo`, `foo.bar.baz/foo-bar123.B_az`.
+      * Invalid: `foo`, `vendor.com/foo/`, `vendor.com/foo/bar`.
+
+#### Container Runtime
+
+* `containerRuntime` (array of string, OPTIONAL) indicates that this CDI specification targets only a specific container runtime.
+  If this field is not indicated then container runtimes should consider that the JSON file targets all runtimes.
+  If this field is indicated then container runtimes should only run it if one of the identifiers matches the container runtime's identifier.
+  Possible values (not an exhaustive list): docker, podman, gvisor, lxc
```

Comment:
Is it that a CDI specification with `containerRuntime` takes precedence over one without ?
its not clear to me when reading through this section

Reply:
https://github.com/container-orchestrated-devices/container-device-interface/pull/8